handlers/macrokiosk/macrokiosk.go

```diff
@@ -1,5 +1,227 @@
 package macrokiosk
 
-/*
-GET /handlers/macrokiosk/receive/uuid?from=60124361111&text=Msg&time=2017-05-03%2010:37:40&msgid=26645777571&shortcode=62000&telcoid=4
-*/
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/buger/jsonparser"
+	"github.com/nyaruka/courier"
+	"github.com/nyaruka/courier/gsm7"
+	"github.com/nyaruka/courier/handlers"
+	"github.com/nyaruka/courier/utils"
+	"github.com/nyaruka/gocommon/urns"
+)
+
+var (
+	sendURL                   = "https://www.etracker.cc/bulksms/send"
+	maxMsgLength              = 1600
+	configMacrokioskSenderID  = "macrokiosk_sender_id"
+	configMacrokioskServiceID = "macrokiosk_service_id"
+)
+
+func init() {
+	courier.RegisterHandler(newHandler())
+}
+
+type handler struct {
+	handlers.BaseHandler
+}
+
+func newHandler() courier.ChannelHandler {
+	return &handler{handlers.NewBaseHandler(courier.ChannelType("MK"), "Macrokiosk")}
+}
+
+// Initialize is called by the engine once everything is loaded
+func (h *handler) Initialize(s courier.Server) error {
+	h.SetServer(s)
+	err := s.AddHandlerRoute(h, http.MethodPost, "status", h.receiveStatus)
+	if err != nil {
+		return err
+	}
+	err = s.AddHandlerRoute(h, http.MethodGet, "status", h.receiveStatus)
+	if err != nil {
+		return err
+	}
+	err = s.AddHandlerRoute(h, http.MethodGet, "receive", h.receiveMessage)
+	if err != nil {
+		return err
+	}
+	return s.AddHandlerRoute(h, http.MethodPost, "receive", h.receiveMessage)
```

Comment:
Looking at logs, it appears these are always GET

Reply:
I checked the logs however I think the method is configurable by the user on macrokiosk side

Reply:
Maybe then we just tell them to always use GET in our configuration?

Reply:
Ok